typescript/packages/well-log-viewer/src/components/ColorTableTypes.ts

```diff
@@ -1,9 +1,18 @@
-export interface ColorTable {
+import type { colorTablesObj } from "@emerson-eps/color-tables/dist/component/colorTableTypes";
+export type colorFunctionsObj = {
     name: string;
-    colors: [number, number, number, number][];
-    discrete?: boolean; // default false
+    func: (v: number) => [number, number, number];
+};
 
-    colorNaN?: [number, number, number]; // default "white"
-    colorBelow?: [number, number, number]; // default colorNaN
-    colorAbove?: [number, number, number]; // default colorBelow
+export type ColorFunction = colorTablesObj | colorFunctionsObj;
+
+export function isFunction(colorFunction: ColorFunction | undefined): boolean {
+    if (!colorFunction) return false;
+    return !!(colorFunction as colorFunctionsObj).func;
 }
+
+import PropTypes from "prop-types";
```

Comment:
What component? 
ColorFunctionType is tightly linked with ColorFunction struct and it very convenient when these 2 types are defined side by side to avoid any discrepancies between them.

Reply:
ColorFunction and co (up to line 16) are standalone (ie. not related at all to React).

"prop-types" is for React components. That's why I prefer moving the React related part to somewhere else, like in typescript/packages/well-log-viewer/src/CommonProps.tsx:

import PropTypes from "prop-types";
export const ColorFunctionType = PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.object,
]);

Reply:
collect all such PropTypes in src\components\CommonPropTypes.ts